SteakBot.Core/EventHandlers/VoiceStateEventHandler.cs

```diff
@@ -1,33 +1,62 @@
 using SteakBot.Core.EventHandlers.Abstraction;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 
 namespace SteakBot.Core.EventHandlers
 {
     internal class VoiceStateEventHandler : IVoiceStateEventHandler
     {
-        public async Task HandleUserVoiceStateUpdatedAsync(SocketUser user, SocketVoiceState leaveState, SocketVoiceState joinState)
+        private const string BendoverwatchMesssageChannel = "bendoverwatch";
+
+        // TODO: get from configuration
+        private readonly IDictionary<string, string> _voiceToMessageChannelMapping = new Dictionary<string, string>
+        {
+            { "shavilandia", "bendoverwatch" },
+            { "starcraft_voice", "zurg_op" },
+            { "random_voice", "parjoli" }
+        };
+
+        public async Task HandleUserVoiceStateUpdatedAsync(SocketUser user, SocketVoiceState oldState, SocketVoiceState newState)
+        {
+            // join
+            if (oldState.VoiceChannel == null && newState.VoiceChannel != null)
+            {
+                await SendMessageAsync(newState, $"{user.Mention} has joined {newState.VoiceChannel.Name}");
+            }
+            // leave
+            else if (oldState.VoiceChannel != null && newState.VoiceChannel == null)
+            {
+                await SendMessageAsync(oldState, $"{user.Mention} has ragequit {oldState.VoiceChannel.Name}");
+            }
+            // change channel
+            else if (oldState.VoiceChannel != null && newState.VoiceChannel != null
+                     && oldState.VoiceChannel.Name != newState.VoiceChannel.Name)
+            {
+                await SendMessageAsync(newState, $"{user.Mention} has moved from {oldState.VoiceChannel.Name} to {newState.VoiceChannel.Name}");
+            }
+        }
+
+        private async Task SendMessageAsync(SocketVoiceState state, string message)
         {
-            if (joinState.VoiceChannel != null)
+            var messageChannelName = GetMessageChannelForVoiceEvent(state.VoiceChannel.Name);
+            var channel = state.VoiceChannel.Guild.Channels.FirstOrDefault(x => x.Name == messageChannelName);
+            if (channel is ISocketMessageChannel messageChannel)
             {
-                // TODO: Perhaps not hardcode this so?
-                var channel = joinState.VoiceChannel.Guild.Channels.FirstOrDefault(x => x.Name == "bendoverwatch");
-                if (channel is ISocketMessageChannel messageChannel)
-                {
-                    await messageChannel.SendMessageAsync($"{user.Mention} has joined {joinState.VoiceChannel.Name}");
-                }
+                await messageChannel.SendMessageAsync(message);
             }
-            else
-            if (leaveState.VoiceChannel != null)
+        }
+
+        private string GetMessageChannelForVoiceEvent(string voiceChannelName)
+        {
+            if (_voiceToMessageChannelMapping.ContainsKey(voiceChannelName))
```

Comment:
Maybe convert this to '?:' operator?

Reply:
I prefer to leave it as is, it looks cleaner than ternary operator to me, if we instead use ternary operator its better to remove the method all together since its used in only 1 place?